src/tests/multihost/ad/conftest.py

```diff
@@ -324,9 +324,13 @@ def _join(membersw=None):
             client_ad.join_ad(ad_dc, ad_password, mem_sw='samba')
         else:
             client_ad.join_ad(ad_dc, ad_password)
+        session_multihost.client[0].run_command(
+            "cp -af /etc/sssd/sssd.conf /etc/sssd/sssd.conf.adjoin")
```

Comment:
This shouldn't be necessary. Any configuration changes to /etc/sssd/sssd.conf before realm join and leave shouldn't be persistent. The command does delete the domain-specific contents of the file. 

[root@thanos ~]# realm join
Password for Administrator: 
[root@thanos ~]# cat /etc/sssd/sssd.conf 

[sssd]
domains = jawkes.com
config_file_version = 2
services = nss, pam

[domain/jawkes.com]
default_shell = /bin/bash
krb5_store_password_if_offline = True
cache_credentials = True
krb5_realm = JAWKES.COM
realmd_tags = manages-system joined-with-adcli 
id_provider = ad
fallback_homedir = /home/%u@%d
ad_domain = jawkes.com
use_fully_qualified_names = True
ldap_id_mapping = True
access_provider = ad



[root@thanos ~]# realm leave
[root@thanos ~]# cat /etc/sssd/sssd.conf 

[sssd]
domains = 
config_file_version = 2
services = nss, pam

Reply:
Actually I had issues leaving realm when the sssd.conf got modified/broken when I was porting ad params suite from bash. I am backing up and restoring the config in each test manually there (The fixture backusssdconf for that got called at a wrong time and realm leave failed so it did not help.)
This will make the code more robust and I would be able to drop the backup/restore code from ad parameters ported suite.

Reply:
I can explain why the bash tests may have given you problems. sssd.conf was configured using a static template from setup. Anyways, understand, LGTM.